pmd-core/src/main/java/net/sourceforge/pmd/RuleSetFactory.java

```diff
@@ -138,7 +144,10 @@ public RuleSetFactory(final RuleSetFactory factory, final boolean warnDeprecated
      * @return An Iterator of RuleSet objects.
      *
      * @throws RuleSetNotFoundException if the ruleset file could not be found
+     *
+     * @deprecated This is apparently only used in code deprecated for removal
```

Comment:
Isn't that used by pmd-doc? And I think, it's used by eclipse-pmd-plugin as well. We probably need to think, whether we can really remove it or whether we need to provide an alternative.
The method sounds more like some utility/helper method, to get all rules of all languages on the classpath.

Reply:
No you're right, this is a mistake. I think this may be useful, but it should not be in RulesetFactory